addons/binding/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/handler/TeslaHandler.java

```diff
@@ -983,99 +973,85 @@ protected boolean establishEventStream() {
 
         @Override
         public void run() {
-            try {
-                if (getThing().getStatus() == ThingStatus.ONLINE) {
-                    if (isAwake()) {
-                        if (establishEventStream()) {
-
-                            String line = null;
-                            try {
-                                line = eventBufferedReader.readLine();
-                            } catch (Exception e) {
-                                logger.error("Event Stream : An exception occurred while reading events : '{}'",
-                                        e.getMessage());
-                                isEstablished = false;
-                            }
-
-                            Date date = new Date();
+            while (true) {
```

Comment:
Is there a sleep statement that I am overlooking? Otherwise this looks to me as if we are running in an endless loop causing 100% CPU in case the status is not ONLINE.